DataScience/Experimentation.py

```diff
@@ -265,19 +271,19 @@ def generate_predictions_files(log_fp, policies):
             marginal_features = marginal_tmp
 
     print("\n*********** SETTINGS ******************")
-    print("Parallel processes: {}".format(n_proc))
+    print("Parallel processes: {}".format(args.n_proc))
     print()
     print("Base command + log file: {}".format(base_command))
     print()
-    print('Learning rates: ['+', '.join(map(str,learning_rates))+']')
-    print('L1 regularization: ['+', '.join(map(str,regularizations))+']')
-    print('Power_t rates: ['+', '.join(map(str,power_t_rates))+']')
+    print('Learning rates: ['+', '.join(map(str,args.learning_rates))+']')
+    print('L1 regularization: ['+', '.join(map(str,args.regularizations))+']')
+    print('Power_t rates: ['+', '.join(map(str,args.power_t_rates))+']')
     print()
     print("Shared feature namespaces: " + str(shared_features))
     print("Action feature namespaces: " + str(action_features))
     print("Marginal feature namespaces: " + str(marginal_features))
     print("***************************************")
-    if input('Press ENTER to start (any other key to exit)...' ) != '':
+    if __name__ == '__main__' and input('Press ENTER to start (any other key to exit)...' ) != '':
```

Comment:
Looking at the following link, this can happen if some one is calling Experimentation.py directly instead of calling Experimentation.main() from somewhere else. That said, I am not experienced in python. If you think there are any other concerns, I am happy to address.

[https://medium.freecodecamp.org/whats-in-a-python-s-name-506262fe61e8](https://medium.freecodecamp.org/whats-in-a-python-s-name-506262fe61e8) 

---
In reply to: [279962265](https://github.com/Microsoft/mwt-ds/pull/132#discussion_r279962265) [](ancestors = 279962265)

Reply:
If `__name__` is inside a `def` it will never be `__main__` no matter how you call the parent script

Reply:
You're right. Your usage is the right one!